core/src/main/java/io/grpc/internal/ManagedChannelImplBuilder.java

```diff
@@ -636,9 +637,14 @@ public ManagedChannel build() {
   // TODO(zdapeng): FIX IT
   @VisibleForTesting
   List<ClientInterceptor> getEffectiveInterceptors() {
-    List<ClientInterceptor> effectiveInterceptors =
-        new ArrayList<>(this.interceptors);
-    if (statsEnabled) {
+    List<ClientInterceptor> effectiveInterceptors = new ArrayList<>(this.interceptors);
+    boolean isGlobalInterceptorsSet = InternalGlobalInterceptors.isGlobalInterceptorsTracersSet();
+    List<ClientInterceptor> globalClientInterceptors =
```

Comment:
This is fine, but is pretty subtle and there's no comment about what's happening. Would it be better for `getClientInterceptors()` to return `null` if the global interceptors were unset? That'd be as easy as initializing the fields to `null` and making sure to set them to non-null in `setInterceptorsTracers()`.

Reply:
CC @DNVindhya . I think in her PR for the implementation of GlobalInterceptors it was implemented this way. I'll change it to `null`

Reply:
Done. PTAL